src/SearchField/SearchFieldAdvanced.jsx

```diff
@@ -0,0 +1,172 @@
+/* eslint-disable max-len */
+import React, { useRef, createContext, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSearch, faTimes } from '@fortawesome/free-solid-svg-icons';
+
+import newId from '../utils/newId';
+
+export const SearchFieldContext = createContext();
+
+const SearchFieldAdvanced = (props) => {
+  const {
+    children,
+    className,
+    screenReaderText,
+    icons,
+    onSubmit,
+    onClear,
+    onChange,
+    onBlur,
+    onFocus,
+    value: initialValue,
+  } = props;
+
+  const [hasFocus, setHasFocus] = useState(false);
+  const [value, setValue] = useState(initialValue);
+
+  const isInitialMount = useRef(true);
+  const inputId = useRef(`${newId('pgn-searchfield-input-')}`);
+  const inputRef = useRef();
+  const submitButtonRef = useRef();
+
+  useEffect(() => {
+    if (isInitialMount.current) {
+      isInitialMount.current = false;
+    } else {
+      onChange(value);
+    }
+  }, [value]);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    onSubmit(value);
+    if (submitButtonRef && submitButtonRef.current) {
+      submitButtonRef.current.focus();
+    }
+  };
+
+  const handleClear = () => {
+    setValue('');
+    onClear();
+    if (inputRef && inputRef.current) {
+      inputRef.current.focus();
+    }
+  };
+
+  const handleFocus = (event) => {
+    setHasFocus(true);
+    onFocus(event);
+  };
+
+  const handleBlur = (event) => {
+    setHasFocus(false);
+    onBlur(event);
+  };
+
+  const handleChange = (event) => {
+    setValue(event.target.value);
+  };
+
+  return (
+    <div
+      className={classNames(
+        'pgn-searchfield',
+        'd-flex border rounded',
```

Comment:
Let's avoid using bootstrap utilities as much as possible. `border` and `rounded` should probably be part of the default search field scss or our edx theme.